File: frontend/TodoFromScratch/app/src/main/java/com/example/todofromscratch/game/Map.java
package com.example.todofromscratch.game;

import java.util.Random;



public class Map {

    private int[][] map;

    public Map() {
        map = new int[10][10];

        // Randomly fill the map
        Random random = new Random();
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                int randNum = random.nextInt(3);
                map[i][j] = randNum;
            }
        }
    }

    // Override toString for the map
    @Override
    public String toString() {
        StringBuilder tempStr = new StringBuilder();
        for (int[] row : map) {
            for (int cell : row) {
                tempStr.append(cell).append(" ");
            }
            tempStr.append("\n");
        }
        return tempStr.toString();
    }

    public GameEnum.Terrain move(int x, int y) {
        if (x < 0 || x > 9 || y < 0 || y > 9) {
            return GameEnum.Terrain.INVALID;
        } else {
            return GameEnum.Terrain.FOREST;
        }
    }

    public String getSurrounding(int x, int y) {
        StringBuilder surrounding = new StringBuilder();
        for (int x_row = -3; x_row <= 3; x_row++) {
            for (int y_col = -3; y_col <= 3; y_col++) {

                if (x_row + x < 0 || x_row + x > 9 || y_col + y < 0 || y_col + y > 9) {
                    continue;
                } else {
                    surrounding.append(map[x_row + x][y_col + y]).append(" ");
                }
            }
            surrounding.append('\n');
        }
        return surrounding.toString();
    }

    public static void main(String[] args) {
        Map map = new Map();
        System.out.println(map);
        System.out.println(map.getSurrounding(5, 5));
        System.out.println(map.move(5, 5));
        System.out.println(map.move(10, 10));
        System.out.println(map.move(-1, -1));
        System.out.println(map.move(0, 0));
        System.out.println(map.move(9, 9));
        System.out.println(map.move(0, 9));
        System.out.println(map.move(9, 0));
    }
}

File: game_prototype/gamePython/Map.py

import random
import GameEnum
from colorama import Fore, Back, Style
class Map:

    map = [[0 for i in range(10)] for j in range(10)]
    
    def __init__(self):
        #randomly fill the map
        for i in range(10):
            for j in range(10):

                if random.randint(0, 1) == 0:
                    self.map[i][j] = 0
                elif random.randint(0, 1) == 1:
                    self.map[i][j] = 1
                else:
                    self.map[i][j] = 2

    #override print for the map
    def __str__(self):
        tempStr = ""
        for i in self.map:
            for j in i:
                tempStr += str(j) + " "
            tempStr += "\n"
        return tempStr
    def move(self, x, y):
        if (x < 0) or (x > 9) or (y < 0) or (y > 9):
            return GameEnum.Terrain.INVALID
        else:
            return GameEnum.Terrain.FOREST

    def getSurrounding(self, x, y):
        surrounding = ""
        for x_row in range(-3,4):
            for y_col in range(-3,4):

                if x_row + x < 0 or x_row + x > 9 or y_col + y < 0 or y_col + y > 9:
                    continue
                else:
                    surrounding += str(self.map[x_row + x][y_col + y]) + " "
            surrounding += '\n'
        return surrounding

if __name__ == "__main__":
    map = Map()
    print(map)
    print(map.getSurrounding(5, 5))
    print(map.move(5, 5))
    print(map.move(10, 10))
    print(map.move(-1, -1))
    print(map.move(0, 0))
    print(map.move(9, 9))
    print(map.move(0, 9))
    print(map.move(9, 0))
    